fix flaky test `test_process_kill_no_matching_pid`

### First check

- [X] I am a contributor to the Prefect codebase

### Description

This test inconsistently fails like

```
self = Process(type='process', env={}, labels={}, name=None, command=['noop'], stream_output=True, working_dir=None)
infrastructure_pid = 'runner:12345', grace_seconds = 30

    async def kill(self, infrastructure_pid: str, grace_seconds: int = 30):
        hostname, pid = _parse_infrastructure_pid(infrastructure_pid)
    
        if hostname != socket.gethostname():
            raise InfrastructureNotAvailable(
                f"Unable to kill process {pid!r}: The process is running on a different"
                f" host {hostname!r}."
            )
    
        # In a non-windows environment first send a SIGTERM, then, after
        # `grace_seconds` seconds have passed subsequent send SIGKILL. In
        # Windows we use CTRL_BREAK_EVENT as SIGTERM is useless:
        # https://bugs.python.org/issue26350
        if sys.platform == "win32":
            try:
                os.kill(pid, signal.CTRL_BREAK_EVENT)
            except (ProcessLookupError, WindowsError):
                raise InfrastructureNotFound(
                    f"Unable to kill process {pid!r}: The process was not found."
                )
        else:
            try:
>               os.kill(pid, signal.SIGTERM)
E               PermissionError: [Errno 1] Operation not permitted

src/prefect/infrastructure/process.py:196: PermissionError

---

FAILED tests/infrastructure/test_process.py::test_process_kill_no_matching_pid - PermissionError: [Errno 1] Operation not permitted
```

### Impact

flaky tests are a distraction

### Additional context

_No response_

--- a/tests/infrastructure/test_process.py
+++ b/tests/infrastructure/test_process.py
@@ -17,6 +17,8 @@ from prefect.exceptions import InfrastructureNotAvailable, InfrastructureNotFoun
 from prefect.infrastructure.process import Process
 from prefect.testing.utilities import AsyncMock
 
+CTRL_BREAK_EVENT = signal.CTRL_BREAK_EVENT if sys.platform == "win32" else None
+
 
 @pytest.fixture
 def mock_open_process(monkeypatch):
@@ -255,14 +257,32 @@ async def test_process_kill_mismatching_hostname(monkeypatch):
     os_kill.assert_not_called()
 
 
-@pytest.mark.flaky(max_runs=2)
-async def test_process_kill_no_matching_pid(monkeypatch):
-    infrastructure_pid = f"{socket.gethostname()}:12345"
+@pytest.mark.parametrize(
+    "platform, expected_signal, skip_if",
+    [
+        ("win32", CTRL_BREAK_EVENT, sys.platform != "win32"),
+        ("linux", signal.SIGTERM, False),
+        ("darwin", signal.SIGTERM, False),
+    ],
+)
+async def test_process_kill_no_matching_pid(
+    monkeypatch, platform, expected_signal, skip_if
+):
+    if skip_if:
+        pytest.skip("Skipping Windows test on non-Windows platforms.")
 
+    monkeypatch.setattr(sys, "platform", platform)
+
+    mock_kill = MagicMock(side_effect=ProcessLookupError)
+    monkeypatch.setattr("os.kill", mock_kill)
+
+    infrastructure_pid = f"{socket.gethostname()}:12345"
     process = Process(command=["noop"])
 
     with pytest.raises(InfrastructureNotFound):
         await process.kill(infrastructure_pid=infrastructure_pid)
+
+    mock_kill.assert_called_once_with(12345, expected_signal)
 
 
 @pytest.mark.skipif(

